src/ManualInput/Input.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { forwardRef, useImperativeHandle, useRef, useState } from 'react';
+import { StyleSheet, Text, TextInput, TouchableOpacity, View } from 'react-native';
+
+import { ManualInputDefault } from '../types';
+const obfusChars = ['/', ':'];
+type Props = {
+    defaultValue: ManualInputDefault;
+    onInputChange: (text: string) => void;
+    maxLength?: number;
+};
+export type FocusHandle = {
+    focus: () => void;
+};
+const Input = forwardRef<FocusHandle, Props>(({ defaultValue, onInputChange, maxLength }, ref) => {
+    const [value, setValue] = useState('');
+    const textInputRef = useRef<TextInput>(null);
+
+    useImperativeHandle(ref, () => ({
+        focus: () => {
+            textInputRef?.current?.focus();
+        },
+    }));
+    const minWidth = defaultValue === ManualInputDefault.LongDate ? 130 : 94;
+
+    const handleChange = (text: string) => {
+        setValue(text);
+        onInputChange(text);
+    };
+
+    const renderMaskedText = (text: string) => {
+        let val = defaultValue.split('');
+        let j = 0;
+        for (let i = 0; i < defaultValue.length; i++) {
+            if (obfusChars.includes(defaultValue[i])) {
+                val[i] = defaultValue[i];
+            } else {
+                val[i] = text[j] ?? defaultValue[i];
+                j++;
+            }
+        }
+        return <Text style={styles.maskedText}>{val.join('')}</Text>;
+    };
+
+    return (
+        <View style={styles.container}>
+            <TouchableOpacity
+                onPress={() => {
+                    textInputRef.current?.focus();
+                }}
+                style={styles.maskedTextContainer}>
+                <Text style={styles.maskedText}>{renderMaskedText(value)}</Text>
+            </TouchableOpacity>
+
+            <TextInput
+                ref={textInputRef}
+                caretHidden={true}
+                value={value}
+                onChangeText={handleChange}
+                style={[styles.textInputStyle, { minWidth }]}
+                maxLength={maxLength}
+                keyboardType="number-pad"
+            />
+        </View>
+    );
+});
+
+const styles = StyleSheet.create({
+    container: {
+        backgroundColor: '#000000',
+        position: 'relative',
+        height: 48,
+    },
+    textInputStyle: {
+        color: 'transparent',
+        backgroundColor: '#131319',
+        paddingHorizontal: 12,
+        borderRadius: 8,
+        paddingVertical: 14.5,
+    },
+    maskedTextContainer: {
+        position: 'absolute',
+        top: 0,
+        left: 0,
+        height: '100%',
+        width: '100%',
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+        paddingHorizontal: 12,
+        zIndex: 1,
+    },
+    maskedText: {
+        color: '#8A8A8F',
```

Comment:
This is a library, so user should be allow to modify color, background color, but since it is hard coded, it can't